frontend/packages/ceph-storage-plugin/src/components/modals/storage-class-dropdown.tsx

```diff
@@ -14,18 +15,30 @@ export const OCSStorageClassDropdown: React.FC<OCSStorageClassDropdownProps> = (
 
 const StorageClassDropdown = (props: any) => {
   const scConfig = _.cloneDeep(props);
+  const [infrastructure, setInfrastructure] = React.useState<K8sResourceKind>();
+  const [infrastructureError, setInfrastructureError] = React.useState();
   /* 'S' of Storage should be Capital as its defined key in resourses object */
   const scLoaded = _.get(scConfig.resources.StorageClass, 'loaded');
-  const scData = _.get(scConfig.resources.StorageClass, 'data', []) as K8sResourceKind[];
+  const scData = _.get(scConfig.resources.StorageClass, 'data', []) as StorageClassResourceKind[];
+
+  React.useEffect(() => {
+    const fetchInfrastructure = async () => {
+      try {
+        const infra = await k8sGet(InfrastructureModel, 'cluster');
+        setInfrastructure(infra);
+      } catch (error) {
+        setInfrastructureError(error);
+      }
+    };
+    fetchInfrastructure();
```

Comment:
IMO we can directly execute the code instead of defining in separate function and the calling it immediately. Also I don't see that `fetchInfrastructure` is getting used anywhere.

Reply:
Await needs async function.